src/storage/local.rs

```diff
@@ -53,10 +54,22 @@ impl Backend for LocalBackend {
         })
     }
     
-    fn upload_artifact_dir(&self, _name: &str, _version: u32, _env: Option<&str>) -> LalResult<()> {
+    fn upload_artifact_dir(&self, name: &str, version: u32, env: Option<&str>) -> LalResult<()> {
         // this fn basically assumes all the sanity checks have been performed
         // files must exist and lockfile must be sensible
-        let _artdir = Path::new("./ARTIFACT");
+        let artifactdir = Path::new("./ARTIFACT");
+        let tarball = artifactdir.join(format!("{}.tar.gz", name));
+        let lockfile = artifactdir.join("lockfile.json");
+
+        // prefix with environment
+        let prefix = env.map(|s| format!("env/{}/", s)).unwrap_or_else(|| "global/".into());
```

Comment:
Forgot that env is still an option here due to the global join location :/
This is a very nice way of doing the ternary though.